### ISSUE TITLE
[syncfusion_flutter_calendar] raiseCalenarTapCallback should check if notes exist before calling.

### ISSUE THREAD

**Issue by ericpark** [syncfusion/flutter-widgets#1731]
in calendar_view_helper.dart, the appointment notes property is used even though the callback allows for dynamic. 

This can be fixed with 
``` if (appointment.hasOwnProperty('notes'))```

calendar_view_helper.dart
```       
static void raiseCalendarTapCallback(
      SfCalendar calendar,
      DateTime? date,
      List<dynamic>? appointments,
      CalendarElement element,
      CalendarResource? resource) {
    if (appointments != null && appointments.isNotEmpty) {
      for (final dynamic appointment in appointments) {
        String? notes = appointment.notes;
        if (notes != null && notes.isNotEmpty) {
          notes = notes.replaceAll('isOccurrenceAppointment', '');
          appointment.notes = notes;
        }
      }
    }
    calendar.onTap!(CalendarTapDetails(appointments, date, element, resource));
  }
```

This should be: 

```       
static void raiseCalendarTapCallback(
      SfCalendar calendar,
      DateTime? date,
      List<dynamic>? appointments,
      CalendarElement element,
      CalendarResource? resource) {
    if (appointments != null && appointments.isNotEmpty) {
      for (final dynamic appointment in appointments) {
        if (appointment.hasOwnProperty('notes')) {
          String? notes = appointment.notes;
          if (notes != null && notes.isNotEmpty) {
            notes = notes.replaceAll('isOccurrenceAppointment', '');
            appointment.notes = notes;
          }
        }
      }
    }
    calendar.onTap!(CalendarTapDetails(appointments, date, element, resource));
  }
```

**Comment by Yuvaraj-Gajaraj** [comment]
The reported issue has been already fixed and rolled out in our Volume 1 2024 main release. To avoid this please upgrade your package to the latest version mentioned below.

Latest package: https://pub.dev/packages/syncfusion_flutter_datepicker/versions/25.1.35

**Comment by thobiassilva** [comment]
Solved on version 25.1.35

https://help.syncfusion.com/flutter/release-notes/v25.1.35?type=all#calendar
- #FB51076 - Now, the notes method not found exception will not be thrown when tapping on appointment.
